### ISSUE TITLE
EngineAdapters inconsistently report exceptions

### ISSUE THREAD

**Issue by saper** [sass/sass-spec#497]
To demonstrate this, please use https://github.com/saper/sass-spec/tree/repro_497 branch (it fixes #495, #496 and includes #494 as well as sample tests for `@error` and `@warn`).

Ruby Sass version [3.4.18.4ef8e31 (Selective Steve)](https://github.com/sass/sass/commit/4ef8e3167985ace91b2105916756bd93c5d7bba6).

SassC version is `sassc: 3.3.0-beta1, libsass: 3.3.0-beta2-1-g1efd, sass2scss: 1.0.3`

### Command to test:
    ruby -I <path-to-ruby-sass> sass-spec.rb -s spec/misc/error-directive/
### Results:
Standard error: empty
Effect on standard output: https://gist.github.com/a4432027a9f3545531a8
**A full exception with a backtrace is returned formatted as a CSS comment**

### Command to test:
    ruby sass-spec.rb -c <path-to-ruby-sass>/bin/sass -s spec/misc/error-directive/
### Results:
Standard error: https://gist.github.com/3c6133b710852b965cb1
````
Error: Buckle your seatbelt Dorothy, 'cause Kansas is going bye-bye
        on line 1 of /home/saper/sw/libsass/sass-spec/spec/misc/error-directive/input.scss
  Use --trace for backtrace.
````
Standard output: empty

### Command to test:
    ruby sass-spec.rb -c <path-to-sassc>/bin/sassc -s spec/misc/error-directive/
### Results:
Standard error: https://gist.github.com/5ac0b9e18252c4906593
````
Error: Buckle your seatbelt Dorothy, 'cause Kansas is going bye-bye
        on line 1 of spec/misc/error-directive/input.scss
>> @error "Buckle your seatbelt Dorothy, 'cause Kansas is going bye-bye"
   ^
````

Standard output: empty

Clearly a backtrace should not be used to test against errors. We need to make sure the exception is dumped in one place (probably standard output) and that the format is the same in all cases.

**Comment by chriseppstein** [comment]
There's a ton of error normalization happening now. In some cases, it could be better, I think some error messages get truncated

**Comment by chriseppstein** [comment]
Pretty sure this is basically fixed since e88e06ffa9198514e29f1dc236bb1618c59fad3a.
